Redesigned land prison

Guldane2 opened the pull request:
#### Summary

SUMMARY: Content "Redesigned land prison"

#### Purpose of change

Due to the changed crafting recipes for the improvised lockpick, the only feasible method (barring extremely lucky drops to allow the crafting of lockpicks) of escaping the existing prison layout was to build a homewrecker, which was extremely tedious on top of an already fairly set start path. 

I also noticed that the existing prison layout did not have a roof, and I thought it was somewhat unrealistic.

#### Describe the solution

Redesigned the prison layout to both be a bit more realistic and allow for the possibility of crafting lockpicks. Added a roof.

#### Describe alternatives you've considered

I considered making more minor edits, such as just adding a few items to make crafting lockpicks a bit more reliable, but decided against.

#### Testing

Ran through a number of created worlds, confirmed I was able to successfully survive and escape. Additional viewpoints welcome.

#### Additional context

Night-Pryanik commented:
Could you please list the changes you made, in details? Also, screenshots would be welcome.

Guldane2 commented:
I basically redesigned the entire prison. The only thing that's still mostly the same is the hidden lab variant's basement entrance.

![Prison3f](https://user-images.githubusercontent.com/66560663/84923718-2cfee100-b08d-11ea-8691-bbf110505632.png)
![Prison2f](https://user-images.githubusercontent.com/66560663/84923723-2f613b00-b08d-11ea-97b8-34b10cd74ad9.png)
![Prison1f](https://user-images.githubusercontent.com/66560663/84923728-312afe80-b08d-11ea-83ee-61c11a9b96b3.png)
![Prison1b](https://user-images.githubusercontent.com/66560663/84923739-338d5880-b08d-11ea-86b5-a2f73351ec5b.png)

anothersimulacrum commented on data/json/mapgen/prison_1.json:
```diff
@@ -14,77 +14,77 @@
+        " |         s f      3333                    f  ss                     | ",
+        " |    |---|s f                  X  X  X  X  f  ss            |---|    | ",
+        " |    |>^<|s f                              f  ss            |<^>|    | ",
+        " |    |^^^+s f3                             f  ssssssssssssss+^^^|    | ",
```
Why would the guard towers have normal wooden doors when other secure areas of the prison have metal ones?

Guldane2 replied:
I changed them from metal to wood when I added the maintenance tunnels, reasoning that they'd be slightly lower security due to needing more regular access (and it somewhat defeats the purpose of there being a chance for the tools needing to craft lockpicks being in the maintenance tunnels if you need lockpicks to access the tunnels to begin with).

anothersimulacrum replied:
Then the guard towers shouldn't have the maintenance tunnels, those should probably be accessed from the main building.

Guldane2 replied:
Maintenance tunnels now originate from the guard barracks closer to the outside, guard towers have metal doors again.

Guldane2 commented:
I changed them from metal to wood when I added the maintenance tunnels, reasoning that they'd be slightly lower security due to needing more regular access (and it somewhat defeats the purpose of there being a chance for the tools needing to craft lockpicks being in the maintenance tunnels if you need lockpicks to access the tunnels to begin with).

nexusmrsep commented:
It feels quite unsecured on the zero level - prison yard with the workout equipment is enclosed, but there is free passage between buildings, free access to administration building, no fences between, no locks to let people in/out in a secure manner. No iconic barb wire anywhere- even just for looks? Other things I've noticed - towers seem to have no eyesight on the western wall - nothing guards that side. What's the reason behind an autodoc in a prison?

Guldane2 commented:
There's bars between the cells and the exits from the cell blocks to the wider yard, but that it's not 100% secure was an intentional design decision on my part; this is not intended to be a supermax prison, but instead more low to medium security. We already have the island prison being a higher security prison. I would expect during normal operation there were regular patrols around the perimeter wall. 

There is actually barbed wire on top of the perimeter wall.

The autodoc is only in the hidden lab variant, and it's behind a science ID door with stairs going down into a lab. It does not exist in the normal version.

I-am-Erk commented:
Looking at real life prison aerial photos, I'd say the main change is that you should block off areas prisoners are expected to walk with more fencing, to prevent free wandering about. Fences leading from the sleeping buildings to the mess building, and blocking off where prisoners are able to travel to get to the rec area. There should only be a single access point to the rec  yard.

It's a bit small but that is the constraint you're working in. We can look at making a bigger regional correction center some day when we've fixed other maps up more.

Guldane2 commented:
Additional fencing added to section the prison.
